From 2ad415015f61b4ee3797d0806a4f490da2a69388 Mon Sep 17 00:00:00 2001
From: Chris Blake <chrisrblake93@gmail.com>
Date: Mon, 19 Nov 2018 16:14:36 -0600
Subject: [PATCH 1/4] ARM: dts: sunxi-h3-h5: Add even more CPU OPP for H3/H5

While we bump the voltages down a tad (based on sunxi code dump from
friendlyarm h5 boards), we also add new frequencies.
---
 arch/arm/boot/dts/sunxi-h3-h5.dtsi           | 92 ++++++++++++++++++++++++----
 arch/arm64/boot/dts/allwinner/sun50i-h5.dtsi |  2 +-
 2 files changed, 80 insertions(+), 14 deletions(-)

diff --git a/arch/arm/boot/dts/sunxi-h3-h5.dtsi b/arch/arm/boot/dts/sunxi-h3-h5.dtsi
index c776d6f..74865bf 100644
--- a/arch/arm/boot/dts/sunxi-h3-h5.dtsi
+++ b/arch/arm/boot/dts/sunxi-h3-h5.dtsi
@@ -112,79 +112,145 @@
 
 		opp@120000000 {
 			opp-hz = /bits/ 64 <120000000>;
-			opp-microvolt = <1040000 1040000 1300000>;
+			opp-microvolt = <940000 940000 940000>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
 		};
 
 		opp@240000000 {
 			opp-hz = /bits/ 64 <240000000>;
-			opp-microvolt = <1040000 1040000 1300000>;
+			opp-microvolt = <940000 940000 940000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+		};
+
+		opp@312000000 {
+			opp-hz = /bits/ 64 <312000000>;
+			opp-microvolt = <940000 940000 940000>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
 		};
 
 		opp@480000000 {
 			opp-hz = /bits/ 64 <480000000>;
-			opp-microvolt = <1040000 1040000 1300000>;
+			opp-microvolt = <940000 940000 940000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+		};
+
+		opp@528000000 {
+			opp-hz = /bits/ 64 <528000000>;
+			opp-microvolt = <940000 940000 940000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+		};
+
+		opp@624000000 {
+			opp-hz = /bits/ 64 <624000000>;
+			opp-microvolt = <960000 960000 960000>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
 		};
 
 		opp@648000000 {
 			opp-hz = /bits/ 64 <648000000>;
-			opp-microvolt = <1040000 1040000 1300000>;
+			opp-microvolt = <970000 970000 970000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+		};
+
+		opp@672000000 {
+			opp-hz = /bits/ 64 <672000000>;
+			opp-microvolt = <970000 970000 970000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+		};
+
+		opp@720000000 {
+			opp-hz = /bits/ 64 <720000000>;
+			opp-microvolt = <970000 970000 970000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+		};
+
+		opp@768000000 {
+			opp-hz = /bits/ 64 <768000000>;
+			opp-microvolt = <980000 980000 980000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+		};
+
+		opp@792000000 {
+			opp-hz = /bits/ 64 <792000000>;
+			opp-microvolt = <1000000 1000000 1000000>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
 		};
 
 		opp@816000000 {
 			opp-hz = /bits/ 64 <816000000>;
-			opp-microvolt = <1100000 1100000 1300000>;
+			opp-microvolt = <1000000 1000000 1000000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+		};
+
+		opp@864000000 {
+			opp-hz = /bits/ 64 <864000000>;
+			opp-microvolt = <1040000 1040000 1040000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+		};
+
+		opp@912000000 {
+			opp-hz = /bits/ 64 <912000000>;
+			opp-microvolt = <1050000 1050000 1050000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+		};
+
+		opp@936000000 {
+			opp-hz = /bits/ 64 <936000000>;
+			opp-microvolt = <1060000 1060000 1060000>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
 		};
 
 		opp@960000000 {
 			opp-hz = /bits/ 64 <960000000>;
-			opp-microvolt = <1200000 1200000 1300000>;
+			opp-microvolt = <1080000 1080000 1080000>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
 		};
 
 		opp@1008000000 {
 			opp-hz = /bits/ 64 <1008000000>;
-			opp-microvolt = <1200000 1200000 1300000>;
+			opp-microvolt = <1100000 1100000 1100000>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
 		};
 
 		opp@1056000000 {
 			opp-hz = /bits/ 64 <1056000000>;
-			opp-microvolt = <1320000 1320000 1320000>;
+			opp-microvolt = <1150000 1150000 1150000>;
+			clock-latency-ns = <244144>; /* 8 32k periods */
+		};
+
+		opp@1080000000 {
+			opp-hz = /bits/ 64 <1080000000>;
+			opp-microvolt = <1160000 1160000 1160000>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
 		};
 
 		opp@1104000000 {
 			opp-hz = /bits/ 64 <1104000000>;
-			opp-microvolt = <1320000 1320000 1320000>;
+			opp-microvolt = <1170000 1170000 1170000>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
 		};
 
 		opp@1152000000 {
 			opp-hz = /bits/ 64 <1152000000>;
-			opp-microvolt = <1320000 1320000 1320000>;
+			opp-microvolt = <1200000 1200000 1200000>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
 		};
 
 		opp@1200000000 {
 			opp-hz = /bits/ 64 <1200000000>;
-			opp-microvolt = <1320000 1320000 1320000>;
+			opp-microvolt = <1240000 1240000 1240000>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
 		};
 
 		opp@1224000000 {
 			opp-hz = /bits/ 64 <1224000000>;
-			opp-microvolt = <1340000 1340000 1340000>;
+			opp-microvolt = <1260000 1260000 1260000>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
 		};
 
 		opp@1248000000 {
 			opp-hz = /bits/ 64 <1248000000>;
-			opp-microvolt = <1340000 1340000 1340000>;
+			opp-microvolt = <1280000 1280000 1280000>;
 			clock-latency-ns = <244144>; /* 8 32k periods */
 		};
 
diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h5.dtsi b/arch/arm64/boot/dts/allwinner/sun50i-h5.dtsi
index a5b196f..cfec115 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h5.dtsi
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h5.dtsi
@@ -58,7 +58,7 @@
 			clock-frequency = <1200000000>;
 			#cooling-cells = <2>;
 			cooling-min-level = <0>;
-			cooling-max-level = <15>;
+			cooling-max-level = <26>;
 		};
 
 		cpu@1 {
-- 
2.7.4

